lib/visitIdentifierNodes.js

```diff
@@ -53,7 +53,7 @@ function normalizeNode(node, context) {
   }
 
   const isAssignment = KEYS_USED_FOR_ASSIGNMENT.has(key) ||
-    (key === 'key' && parent.parent.type === 'ObjectPattern') ||
+    (key === 'key' && !parent.computed && parent.parent.type === 'ObjectPattern') ||
```

Comment:
Can you come up with a test case demonstrating what this is supposed to fix?

Reply:
added tests